control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go

```diff
@@ -0,0 +1,467 @@
+package registryoperator
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"text/template"
+
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/kas"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/metrics"
+	"github.com/openshift/hypershift/support/util"
+)
+
+const (
+	operatorName         = "cluster-image-registry-operator"
+	workerNamespace      = "openshift-image-registry"
+	workerServiceAccount = "cluster-image-registry-operator"
+	tokenFile            = "token"
+	metricsPort          = 60000
+
+	startScriptTemplateStr = `#!/bin/bash
+set -euo pipefail
+
+while true; do
+   if [[ -f {{ .TokenDir }}/token ]]; then
+      break
+   fi
+   echo "Waiting for client token"
+   sleep 2
+done
+
+echo "{{ .WorkerNamespace }}" > "{{ .TokenDir }}/namespace"
```

Comment:
Nit: We can get the namespace through the downwards api and that in turn would make the template variables all static and allow us to render it in an `init` or `var` declaration

Reply:
Actually all vars are static... I will change to generating this in an init func.